bode/bode/models/task_relations_actions.py

```diff
@@ -51,6 +54,54 @@ def get_relation_types(task_id):
     return {map_to_related_task_schema(relation, task_id) for relation in relations_lhs + relations_rhs}
 
 
+def get_transitive_interchangable_related_tasks(task_id, filters=list()):
+    from bode.models.task import Task
+
+    def get_distinct_uuids(uuids):
+        return list(set(uuids))
+
+    def get_relation_graph(relations):
+        graph = Graph(len(all_interchangable_tasks))
+        uuids = []
+
+        for relation in relations:
+            first_task_id, second_task_id = relation.first_task_id, relation.second_task_id
+            graph.add_edge(str(first_task_id), str(second_task_id))
+            uuids.append(str(first_task_id))
+            uuids.append(str(second_task_id))
+
+        return graph, uuids
+
+    def group_tasks_subsets():
+        task_parent = parents[task_id] if task_id in parents.keys() and parents[task_id] != -1 else task_id
+        tasks_subset = []
```

Comment:
```suggestion
    def group_tasks_subsets():
        task_parent = graph.find_union.find(task_id) if task_id in parents.keys() else task_id
        tasks_subset = []
```

We can't be sure that `parents[task_id]` is root of this task.